### PR TITLE
add func InitUndistortRectifyMap

### PR DESCRIPTION
**Pull request by linyongzuo**

### CODE REVIEW

**Comment by linyongzuo** [comment]
@deadprogram

File: calib3d.go [hunk 1]
```diff
@@ -66,3 +66,27 @@ func FisheyeUndistortImageWithParams(distorted Mat, undistorted *Mat, k, d, knew
 	}
 	C.Fisheye_UndistortImageWithParams(distorted.Ptr(), undistorted.Ptr(), k.Ptr(), d.Ptr(), knew.Ptr(), sz)
 }
+
+// https://docs.opencv.org/master/d9/d0c/group__calib3d.html#ga7dfb72c9cf9780a347fbe3d1c47e5d5a
+func InitUndistortRectifyMap(cameraMatrix Mat, distCoeffs Mat, r Mat, newCameraMatrix Mat, size image.Point, m1type int, map1 Mat, map2 Mat) {
```
Comment (deadprogram) [inline calib3d.go]: Can you please add the GoDoc compatible comment format here?

```go
// InitUndistortRectifyMap [description of what it does...]
//
// For further details, please see:
// ([ink to the OpenCV docs, like you have now]
//
```

Thanks!

File: calib3d.go [hunk 2]
```diff
@@ -66,3 +66,27 @@ func FisheyeUndistortImageWithParams(distorted Mat, undistorted *Mat, k, d, knew
...
+	sz := C.struct_Size{
+		width:  C.int(size.X),
+		height: C.int(size.Y),
+	}
+	C.InitUndistortRectifyMap(cameraMatrix.Ptr(), distCoeffs.Ptr(), r.Ptr(), newCameraMatrix.Ptr(), sz, C.int(m1type), map1.Ptr(), map2.Ptr())
+}
+
+// remember Close the Mat object
+// https://docs.opencv.org/master/d9/d0c/group__calib3d.html#ga7a6c4e032c97f03ba747966e6ad862b1
+func GetOptimalNewCameraMatrixWithParams(cameraMatrix Mat, distCoeffs Mat, imageSize image.Point, alpha float64, newImgSize image.Point, centerPrincipalPoint bool) (Mat, image.Rectangle) {
```
Comment (deadprogram) [inline calib3d.go]: Same as previous comment. Thank you.

Comment (linyongzuo) [reply calib3d.go]: @deadprogram done

File: calib3d.go [hunk 3]
```diff
@@ -66,3 +66,33 @@ func FisheyeUndistortImageWithParams(distorted Mat, undistorted *Mat, k, d, knew
 	}
 	C.Fisheye_UndistortImageWithParams(distorted.Ptr(), undistorted.Ptr(), k.Ptr(), d.Ptr(), knew.Ptr(), sz)
 }
+
+// initUndistortRectifyMap computes the joint undistortion and rectification transformation and represents the result in the form of maps for remap
```
Comment (deadprogram) [inline calib3d.go]: This needs to be capitalized for GoDocs to correctly use them.

File: calib3d.go [hunk 4]
```diff
@@ -66,3 +66,33 @@ func FisheyeUndistortImageWithParams(distorted Mat, undistorted *Mat, k, d, knew
...
+// For more details, please see:
+
```
Comment (deadprogram) [inline calib3d.go]: This cannot be a blank line for GoDocs to correctly use them.

File: calib3d.go [hunk 5]
```diff
@@ -66,3 +66,33 @@ func FisheyeUndistortImageWithParams(distorted Mat, undistorted *Mat, k, d, knew
...
+		width:  C.int(size.X),
+		height: C.int(size.Y),
+	}
+	C.InitUndistortRectifyMap(cameraMatrix.Ptr(), distCoeffs.Ptr(), r.Ptr(), newCameraMatrix.Ptr(), sz, C.int(m1type), map1.Ptr(), map2.Ptr())
+}
+
+// GetOptimalNewCameraMatrixWithParams computes and returns the optimal new camera matrix based on the free scaling parameter.
+// please remember Close the Mat object which returns
+// For more details, please see:
+
```
Comment (deadprogram) [inline calib3d.go]: This also cannot be a blank line.

**Comment by deadprogram** [comment]
Hi @linyongzuo thanks for the changes. I requested a couple more changee because GoDocs is somewhat strict about how it parses the comments.

**Comment by deadprogram** [comment]
Thank you for the updates and contribution @linyongzuo now merging.

**Pull request merged**